mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/navigation/MapboxNavigationEvent.java

```diff
@@ -0,0 +1,169 @@
+package com.mapbox.services.android.telemetry.navigation;
+
+import android.location.Location;
+
+import com.mapbox.services.android.telemetry.constants.TelemetryConstants;
+import com.mapbox.services.android.telemetry.utils.TelemetryUtils;
+
+import java.util.Date;
+import java.util.Hashtable;
+
+/**
+ * Represent events dispatched by the Navigation SDK.
+ */
+public class MapboxNavigationEvent {
+
+  private static final int EVENT_VERSION = 1;
+
+  // Event types
+  public static final String TYPE_TURNSTILE = "navigation.turnstile";
+  public static final String TYPE_DEPART = "navigation.depart";
+  public static final String TYPE_FEEDBACK = "navigation.feedback";
+  public static final String TYPE_ARRIVE = "navigation.arrive";
+  public static final String TYPE_CANCEL = "navigation.cancel";
+
+  // Event keys
+  public static final String KEY_EVENT = "event";
+  public static final String KEY_PLATFORM = "platform";
+  public static final String KEY_OPERATING_SYSTEM = "operatingSystem";
+  public static final String KEY_SDK_IDENTIFIER = "sdkIdentifier";
+  public static final String KEY_SDK_VERSION = "sdkVersion";
+  public static final String KEY_EVENT_VERSION = "eventVersion";
+  public static final String KEY_SESSION_IDENTIFIER = "sessionIdentifier";
+  public static final String KEY_LAT = "lat";
+  public static final String KEY_LNG = "lng";
+  public static final String KEY_GEOMETRY = "geometry";
+  public static final String KEY_CREATED = "created";
+  public static final String KEY_PROFILE = "profile";
+  public static final String KEY_ESTIMATED_DISTANCE = "estimatedDistance";
+  public static final String KEY_ESTIMATED_DURATION = "estimatedDuration";
+  public static final String KEY_REROUTE_COUNT = "rerouteCount";
+  public static final String KEY_VOLUME_LEVEL = "volumeLevel";
+  public static final String KEY_SCREEN_BRIGHTNESS = "screenBrightness";
+  public static final String KEY_APPLICATION_STATE = "applicationState";
+  public static final String KEY_BATTERY_PLUGGED_IN = "batteryPluggedIn";
+  public static final String KEY_BATTERY_LEVEL = "batteryLevel";
+  public static final String KEY_CONNECTIVITY = "connectivity";
+  public static final String KEY_FEEDBACK_TYPE = "feedbackType";
+  public static final String KEY_LOCATIONS_BEFORE = "locationsBefore";
+  public static final String KEY_LOCATIONS_AFTER = "locationsAfter";
+  public static final String KEY_NEW_DISTANCE_REMAINING = "newDistanceRemaining";
+  public static final String KEY_NEW_DURATION_REMAINING = "newDurationRemaining";
+  public static final String KEY_START_TIMESTAMP = "startTimestamp";
+  public static final String KEY_DISTANCE_COMPLETED = "distanceCompleted";
+  public static final String KEY_DISTANCE_REMAINING = "distanceRemaining";
+  public static final String KEY_DURATION_REMAINING = "durationRemaining";
+  public static final String KEY_SECONDS_SINCE_LAST_REROUTE = "secondsSinceLastReroute";
```

Comment:
We've added a few additional keys to look out for in the spec:

General:
* simulation
* originalRequestIdentifier
* requestIdentifier
* originalGeometry
* originalEstimatedDistance
* originalEstimatedDuration

Feedback:
* description
* userId
* screenshot

Reroute:
* newDistanceRemaining
* newDurationRemaining
* newGeometry
* screenshot

Cancel:
* arrivalTimestamp